### DISCUSSION TITLE
Additional features of 2-factor authentication on Mobile App (Native app already exists)

### DISCUSSION THREAD

**Discussion by batrinhit** [invertase/react-native-firebase#5044] [General]
Additional features of 2-factor authentication on Mobile App
Detail:
(Native app already exists)https://cloud.google.com/identity-platform/docs/ios/mfa

**Comment by mikehardy** [comment]
Is this a question, or an issue, or a request ? I don't understand, sorry
I think it probably relates to this, it's open for anyone to work on - #4166

**Reply by batrinhit** [reply]
Currently, on the Web I have enabled 2-factor authentication with OTP SMS
When User Login with username / password, SMS will be sent to phone number previously registered.
The website side has been working well.
But on the App side, the method returns no data related to MFA.
specific Android native: https://cloud.google.com/identity-platform/docs/android/mfa
iOS native: https://cloud.google.com/identity-platform/docs/ios/mfa

**Reply by batrinhit** [reply]
This is the web version of firebase, I also want firebase-react-native/ auth on mobile to have these methods as well.
image: https://ibb.co/cQN3pxQ
<img width="454" alt="Capture" src="https://user-images.githubusercontent.com/26219584/111588655-6ddf7e00-87f6-11eb-8bbf-27cd39dd5fdd.PNG">

**Reply by mikehardy** [reply]
Great, so, without posting images (those don't work across platforms and may not be quoted etc), you should head over to the linked issue which is right on target for you and has some related information. It's open for anyone to implement, we happily merge PRs

**Comment by mikehardy** [comment]
Multifactor with phone released here in v11.3.0
